### PR TITLE
DS real world guide book

### PR DESCRIPTION
**Pull request by Zhaochengxiann**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [ ] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+The default value for the NotebookInstanceSecGroupId parameter is to select the "default security group". The default security group is automatically created when a VPC is created and allows all traffic to pass. If the default security group is selected, SageMaker Notebook instances can be accessed from all IP addresses.
+
+the NotebookInstanceSubnetId parameter is used to specify the subnet where SageMaker Notebook instances are deployed, improving security and availability by deploying them in the specified network location.
+
+## Step 4
+
+After waiting for "create complete", a new notebook will be created in Amazon SageMaker.
+![SageMaker guide 5](../../images/SageMaker-guide-5.png)
+
+Clicking on "Open JupyterLab" will start the job.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md:51]: What's next? This doc is for helping people to finish the experiment of amazon-sagemaker-mlops-workshop. You need to cover all the content and steps for finishing the warm-up section.

**Review by zhangqi444 — CHANGES_REQUESTED** [review]

File: open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Data science in AWS SageMaker 
+
+Amazon SageMaker is a cloud service for machine learning provided by Amazon Web Services (AWS). It offers a complete platform that helps users build, train, and deploy machine learning models, along with a range of tools and algorithms that make machine learning more accessible and scalable.
```
[the remark this answers is not in the document]
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md:3]: ok

File: open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Data science in AWS SageMaker
```
[the remark this answers is not in the document]
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md:1]: ok

File: open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# Data science in AWS SageMaker 
+
+Amazon SageMaker is a cloud service for machine learning provided by Amazon Web Services (AWS). It offers a complete platform that helps users build, train, and deploy machine learning models, along with a range of tools and algorithms that make machine learning more accessible and scalable.
+
+
```
[the remark this answers is not in the document]
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md:5]: ok

File: open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+We will learn how to use a [project](https://github.com/awslabs/amazon-sagemaker-mlops-workshop/) provided by AWS Labs to help users understand the machine learning workflow on the Amazon SageMaker platform. The project covers a series of work steps, from data preparation to model development and deployment, and also presents best practices for managing machine learning workflows in production. The project is mainly divided into the following parts:
```
[the remark this answers is not in the document]
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md:6]: done

File: open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md (hunk 1 above)
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md]: ok

File: open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+1. Data Preparation: In this section, you will learn how to prepare a machine learning dataset and use Amazon SageMaker Data Labeling Tool for data labeling.
+2. Model Development: In this section, you will develop and train machine learning models using Amazon SageMaker Notebook instances, as well as optimize models using Amazon SageMaker Automatic Model Tuning feature.
+3. Model Evaluation and Interpretation: In this section, you will learn how to evaluate and interpret machine learning models using Amazon SageMaker model monitoring and interpretation features.
+4. Model Deployment and Management: In this section, you will learn how to deploy trained models to production environments using Amazon SageMaker model deployment feature, and manage machine learning workflows using Amazon SageMaker management feature.
+
+## Step 1
```
[the remark this answers is not in the document]
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/data-science/data-science-in-AWS-SageMaker.md]: ok

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+![SageMaker guide 7](../../../images/SageMaker-guide-7.png)
+
+The input data is the iris_train.csv previously created in Amazon S3.
+
+
+
+
+
+
+
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:77]: remove the unnecessary empty lines.

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+```{note}
+If using ml.m4.xlarge, the notebook creation will not be successful.
+```
+
+Then, we can add the modified YAML file by clicking on the "Upload a template file" button.
+
+Subsequently, we need to set the NotebookInstanceSecGroupId parameter.
+
+![SageMaker guide 4](../../../images/SageMaker-guide-4.png)
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:36]: change the all file names to have only lowercase.

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+Enter Amazon SageMaker, then enter Notebook instances, we can see that a new notebook is created.
+
+![SageMaker guide 5](../../../images/SageMaker-guide-5.png)
+
+Click "Open JupyterLab" to start the warmup exercises.
+
+### Step 3 - start warm-up section
+
+You can see the four files required to complete the warmup under the path lab/00_Warmup/.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:59]:
```
 `lab/00_Warmup/`
```

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+After running the code for uploading data, the data iris_train.csv and iris_test.csv will be uploaded to the Buckets of Amazon S3.
+
+![SageMaker guide 6](../../../images/SageMaker-guide-6.png)
+
+After running the code for training the model, the model will be trained in Amazon SageMaker. 
+Enter Training jobs to see the model (picture) being trained. 
+
+![SageMaker guide 7](../../../images/SageMaker-guide-7.png)
+
+The input data is the iris_train.csv previously created in Amazon S3.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:70]: \`iris_train.csv\`

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md (hunk 7 above)
Comment (zhangqi444) [reply open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:77]: Please add the acknowledgment section. And add the original license to the head of this file.

e.g. https://open-academy.github.io/machine-learning/prerequisites/python-programming-basics.html

**Review by zhangqi444 — COMMENTED** [review]
Could you please check why the build is failed?

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md (hunk 7 above)
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:77]: done

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md (hunk 9 above)
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:59]: thanks

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md (hunk 10 above)
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:70]: ok

File: open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md (hunk 8 above)
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/amazon-sagemaker-mlops-workshop-warm-up.md:36]: ok

**Pull request closed without merging**